src/test/java/uk/gov/companieshouse/api/testdata/service/impl/AcspProfileServiceImplTest.java

```diff
@@ -83,6 +88,140 @@ void createAcspProfileWithDefaultValues() throws DataException {
+        assertEquals("/authorised-corporate-service-providers/randomId", captured.getLinksSelf());
+        assertEquals(0L, captured.getVersion());
+
+        assertNotNull(captured.getAmlDetails());
+        assertEquals(3, captured.getAmlDetails().size());
+        assertEquals("Supervisory Body 1",
+                captured.getAmlDetails().get(0).getSupervisoryBody());
+        assertEquals("Membership Details 1",
+                captured.getAmlDetails().get(0).getMembershipDetails());
+        assertEquals("Supervisory Body 2",
+                captured.getAmlDetails().get(1).getSupervisoryBody());
+        assertEquals("Membership Details 2",
+                captured.getAmlDetails().get(1).getMembershipDetails());
+        assertEquals("Supervisory Body 3",
+                captured.getAmlDetails().get(2).getSupervisoryBody());
+        assertEquals("Membership Details 3",
+                captured.getAmlDetails().get(2).getMembershipDetails());
+    }
+
+    private static AcspProfile getAcspProfile(AcspProfileSpec spec) {
+        AcspProfile savedProfile = new AcspProfile();
+        savedProfile.setAcspNumber("randomId");
+
+        List<AmlDetails> amlDetailsList = new ArrayList<>();
+        for (AmlSpec amlSpec : spec.getAmlDetails()) {
+            AmlDetails amlDetails = new AmlDetails();
+            amlDetails.setSupervisoryBody(amlSpec.getSupervisoryBody());
+            amlDetails.setMembershipDetails(amlSpec.getMembershipDetails());
+            amlDetailsList.add(amlDetails);
+        }
+        savedProfile.setAmlDetails(amlDetailsList);
+        return savedProfile;
+    }
+
+    private static AcspProfileSpec getAcspProfileSpec() {
+        AcspProfileSpec spec = new AcspProfileSpec();
+        spec.setStatus("active");
+        spec.setType("ltd");
+
+        // Creating multiple AML details entries
+        AmlSpec amlSpec1 = new AmlSpec();
+        amlSpec1.setSupervisoryBody("Supervisory Body 1");
+        amlSpec1.setMembershipDetails("Membership Details 1");
+
+        AmlSpec amlSpec2 = new AmlSpec();
+        amlSpec2.setSupervisoryBody("Supervisory Body 2");
+        amlSpec2.setMembershipDetails("Membership Details 2");
+
+        AmlSpec amlSpec3 = new AmlSpec();
+        amlSpec3.setSupervisoryBody("Supervisory Body 3");
+        amlSpec3.setMembershipDetails("Membership Details 3");
+
+        spec.setAmlDetails(List.of(amlSpec1, amlSpec2, amlSpec3));
+        return spec;
+    }
+
+    @Test
+    void createAcspProfileWithNullAmlDetails() throws DataException {
+        AcspProfileSpec spec = new AcspProfileSpec();
+        spec.setStatus("active");
+        spec.setType("ltd");
+        spec.setAmlDetails(null); // Setting AmlDetails as null
+
+        when(randomService.getString(8)).thenReturn("randomId");
+        AcspProfile savedProfile = new AcspProfile();
+        savedProfile.setAcspNumber("randomId");
+        savedProfile.setAmlDetails(null); // Ensure null is handled correctly
```

Comment:
I don't think we read the aml details from the saved profile object. is this line required?